engine/runtime-parser-processor-tests/src/test/java/org/enso/runtime/parser/processor/test/TestIRProcessorInline.java

```diff
@@ -768,4 +768,36 @@ public String showCode(int indent) {
     assertThat(src, containsString("class JNameGen"));
     assertThat("has getter method for expression", src, containsString("Option<IR> expression()"));
   }
+
+  @Test
+  public void fieldCanBePersistanceReference() {
+    var src =
+        generatedClass(
+            "JName",
+            """
+        import org.enso.runtime.parser.dsl.GenerateIR;
+        import org.enso.runtime.parser.dsl.GenerateFields;
+        import org.enso.runtime.parser.dsl.IRChild;
+        import org.enso.persist.Persistance;
+        import org.enso.compiler.core.IR;
+
+        @GenerateIR
+        public final class JName extends JNameGen {
+          @GenerateFields
+          public JName(@IRChild Persistance.Reference<IR> expression) {
+            super(expression);
+          }
+
+          @Override
+          public String showCode(int indent) {
+            return "";
+          }
+        }
+        """);
+    assertThat(src, containsString("class JNameGen"));
+    assertThat(
+        "has getter method for expression with a different return type",
+        src,
+        containsString("IR expression()"));
```

Comment:
I wonder how `copy` method looks like in this case? Do we share the internal `Persistance.Reference` or do we create a new `Persistance.Reference`?

Reply:
See the [ImportModuleGen.java gist](https://gist.github.com/Akirathan/4685568d9b3710d36e0697304890648e) I have linked from the PR description.